### ISSUE TITLE
chart sankey does not support a->b->a pattern

### ISSUE THREAD

**Issue by shshen** [highcharts/highcharts#8218]
#### Expected behaviour
sankey should be able to render a->b->a 

#### Actual behaviour
sankey render incorrectly to  b->a only.

#### Live demo with steps to reproduce
https://jsfiddle.net/zbnu6w36/

#### Product version
highcharts 6.1, but i think this does not work on all versions.

#### Affected browser(s)
all browser types.

**Comment by pawelfus** [comment]
Hi @shshen 

Thank you for reporting. You are right, circular references are not supported right now.

**Comment by TorsteinHonsi** [comment]
I did some work here, making the layout work in some cases. At least it is better now. But now we need to set chart margins to make room for the links, the next improvement would be to make the layout flow automatically.

* https://jsfiddle.net/highcharts/zbnu6w36/8/
* https://jsfiddle.net/highcharts/zbnu6w36/9/
* https://jsfiddle.net/highcharts/zbnu6w36/7/
* https://jsfiddle.net/highcharts/zbnu6w36/10/

**Comment by hurcy** [comment]
@TorsteinHonsi Could you see this case? 
`a->b, b->c, c->b` doesn't work.
https://jsfiddle.net/m5yjrkv1/

**Comment by TorsteinHonsi** [comment]
Fixed, thanks for reporting! https://jsfiddle.net/highcharts/jfeo8tdy/2/

**Comment by pawelfus** [comment]
Hi @niran-manandhar 

Could you share an example? Note that new version (with fix) is not released, only available on our master branch: 

```
<script src="https://github.highcharts.com/highcharts.js"></script>
<script src="https://github.highcharts.com/modules/sankey.js"></script>
```

**Comment by niran-manandhar** [comment]
**with released** 
https://jsfiddle.net/niran_manandhar/q9b05amd/13/ 
has the stack size error
**with master** 
https://jsfiddle.net/niran_manandhar/tuj874Lz/5/ 
doesnt have the stack size error but chart doesn't seem right

**Comment by TorsteinHonsi** [comment]
You're right, there's still a geometry issue even though the stack size issue is resolved. As a workaround we can override the `translate` function, though there's still something wrong with the green link: https://jsfiddle.net/highcharts/jfeo8tdy/9/

**Comment by niran-manandhar** [comment]
managed to do it with margins and clip:false
https://jsfiddle.net/niran_manandhar/vL913mkb/

**Comment by sruchot** [comment]
Hello,
Someone advise me to comment this issue since it's related to one I'm having at the moment.

When having a reverse flow in your data the chart will display correctly on first render but will behave differently on any resize/update action.

A quick example is set up at: [https://jsfiddle.net/Lelith/7omz94gc/1/](https://jsfiddle.net/Lelith/7omz94gc/1/)

You just have to increase the size of the chart container to reproduce the issue.
Everything looks good on first rendering though.

(original post on HC forum was [https://www.highcharts.com/forum/viewtopic.php?f=9&t=44510](https://www.highcharts.com/forum/viewtopic.php?f=9&t=44510) )

Thanks!

**Comment by applewil** [comment]
Is there a workaround for this issue?

When I plot a->d and b->c->d the arrangement is compressed until I resize the panel triggering a reflow

```
data: [
  ['a', 'd', 1],
  ['b', 'c', 1],
  ['c', 'd', 1]
]
```

Demo: https://jsfiddle.net/applewil/ezjfarh4/6/

How do I get the uncompressed node arrangement when the panel loads?

**Comment by raf18seb** [comment]
Hi @applewil,

As a workaround, you can use the `chart.events.load` event to reflow the chart, see demo: https://jsfiddle.net/BlackLabel/xw24o8kh/

**Comment by megaroeny** [comment]
Any progress on this issue? 6 years ago this was posted! 🤯 Thank you!

**Comment by magdalena-gut** [comment]
Hi @megaroeny,

It looks like the margin workaround should suit your use case: https://jsfiddle.net/BlackLabel/jqw3kune/

That said, I agree - it has been a long time since this issue was addressed. We've now logged it in our backlog and will evaluate it as part of our ongoing prioritization process.

Thank you for your report. We'll share updates here when they become available. In the meantime, feel free to comment if you have any further input or background on the issue.
